src/forEachObj.test.ts

```diff
@@ -70,4 +70,22 @@ describe("typing", () => {
       expectTypeOf(value).toEqualTypeOf<boolean | string>();
     });
   });
+
+  test("union of records", () => {
+    forEachObj(
+      {} as Record<number, string> | Record<string, number>,
+      (value, key) => {
+        expectTypeOf(key).toEqualTypeOf<string>();
+        expectTypeOf(value).toEqualTypeOf<number | string>();
+      },
+    );
+
+    pipe(
+      {} as Record<number, string> | Record<string, number>,
```

Comment:
Similar comments as above, keep a single "data" variable with the type so that both tests are synced and we don't risk changing one without changing the other.
```suggestion
      data,
```